File: 310971/task350/publisher/src/main/java/org/example/task350/publisher/model/Tweet.java
package org.example.task350.publisher.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.OffsetDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "tbl_tweet", schema = "distcomp")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class Tweet extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "writer_id", nullable = false)
    private Writer writer;

    @Column(nullable = false, length = 64, unique = true)
    @Size(min = 2, max = 64)
    private String title;

    @Column(nullable = false, length = 2048)
    @Size(min = 4, max = 2048)
    private String content;

    @Column(nullable = false)
    private OffsetDateTime created;

    @Column(nullable = false)
    private OffsetDateTime modified;

    @OneToMany(mappedBy = "tweet", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<TweetLabel> tweetLabels = new HashSet<>();
}

File: 351003/Gorodko/app/Model/Tweet.cs
namespace Project.Model {
    public class Tweet : BaseEntity {
        public long EditorId { get; set; }
        public Editor? Editor { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public DateTime Created { get; set; }
        public DateTime Modified { get; set; }

        public List<Reaction> Reactions { get; set; } = new();
        public List<Sticker> Stickers { get; set; } = new();
    }
}